platform/stm32f4xx/export/platform/spi_bus.hpp

```diff
@@ -407,14 +464,20 @@ void spi_bus< spi_config >::prepare_tx()
         dma_init.DMA_Memory0BaseAddr = reinterpret_cast< uint32_t >(m_tx.buf);
     }
 
-    dma_init.DMA_BufferSize          = m_tx_size;
+    if (spi_i2s_config::m_interface == spi_i2s_interface::i2s) {
+        dma_init.DMA_MemoryDataSize      = DMA_MemoryDataSize_HalfWord;
+        dma_init.DMA_PeripheralDataSize  = DMA_PeripheralDataSize_HalfWord;
+        dma_init.DMA_BufferSize          = m_tx_size >> 1;
```

Comment:
Substitute to m_tx_size / 2. It shows the right intent of this code.
Division by a constant of power of 2 is well-optimized.